### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by cdavernas**
**What does this PR do?**

Adds the first custom function and adds all documents required by the
 community standards

### CODE REVIEW

**Comment by cdavernas** [comment]
The 'log' function has been written in `python` (rather than `javascript`) after checking which language was most used, according to TIOBE.

File: README.md [hunk 1]
```diff
@@ -20,3 +20,89 @@ For information on how to create a custom function, please refer to the [documen
 
 By providing a centralized repository for custom functions, the "catalog" repository enhances the flexibility and power of the Serverless Workflow DSL, enabling users to create more complex and tailored workflow solutions.
 
+## Structure
+
+Cataloged functions must follow the following structure:
+
+```
+./functions
+    /log                        👈 The name of the custom function. Must be lowercased and contain exclusively alphanumeric characters, with the exception of '-'.
```
Comment (ricardozanini) [inline README.md]:
```suggestion
    /log                        👈 The name of the custom function. It must be lowercase and contain exclusively alphanumeric characters, except for '-'.
```

File: README.md [hunk 2]
```diff
@@ -20,3 +20,89 @@ For information on how to create a custom function, please refer to the [documen
...
+            /function.yaml      👈 The function's definition.
+            /OWNERS.md          👈 Describes the owners/authors of the custom function.
+            /README.md          👈 Describes the custom function.
+        /2.0.0
+            ...
+```
+
+## Usage
+
+Below is an example of how to usea custom function in a Serverless Workflow:
```
Comment (ricardozanini) [inline README.md]:
```suggestion
Below is an example of how to use a custom function in a Serverless Workflow:
```

File: README.md [hunk 3]
```diff
@@ -20,3 +20,89 @@ For information on how to create a custom function, please refer to the [documen
...
+  version: '1.0.0'
+do:
+  - callCatalogedFunction:
+      call: https://github.com/serverlessworkflow/catalog/functions/log/1.0.0 #link to the repository directory that contains the function.yaml of the custom function to call.
+      #call: log:1.0.0 <--- this would also be a correct way to reference the function, as it is registered in the official catalog.
+      with:
+        message: Hello, world!
+```
+
+## Contributing
```
Comment (ricardozanini) [inline README.md:56]: It seems a duplication of `CONTRIBUTION.MD`? Why not have only the file and link it here?

**Review by ricardozanini — APPROVED** [review]
This is insane!

File: README.md (hunk 3 above)
Comment (cdavernas) [reply README.md:56]: Good point!

**Pull request merged**